File: 3.ReverseComplement/ReverseCompliment.go
package main


import (
	"fmt"
)


func ReverseCompliment(pattern string) string {
	compliment_dict := map[byte]byte{
		'A': 'T',
		'T': 'A',
		'C': 'G',
		'G': 'C',
	}
	reverseCompliment := make([]byte, len(pattern))

	for i:=0; i < len(pattern); i++ {
		reverseCompliment[len(pattern)-1-i] = compliment_dict[pattern[i]]
	}

	return string(reverseCompliment)
}


func main(){
	dna := "AAAACCCGGT"
	result := ReverseCompliment(dna)
	fmt.Println(result)
}

File: 3.ReverseComplement/ReverseCompliment.py
def reverse_complement(pattern: str) -> str:
    """Calculate the reverse complement of a DNA pattern."""
    complement_dict = {'A': 'T', 'T': 'A', 'C': 'G', 'G': 'C'}
    complement_translation = str.maketrans(complement_dict)
    return pattern[::-1].translate(complement_translation)



text = "AAAACCCGGT"
print(reverse_complement(text))